ethereum/core/src/main/java/org/hyperledger/besu/ethereum/debug/TraceOptions.java

```diff
@@ -14,6 +14,40 @@
  */
 package org.hyperledger.besu.ethereum.debug;
 
-public record TraceOptions(boolean traceStorage, boolean traceMemory, boolean traceStack) {
-  public static final TraceOptions DEFAULT = new TraceOptions(true, false, true);
+import java.util.Map;
+
+/**
+ * Configuration options for tracing.
+ *
+ * <p>Contains the tracer type and configuration options for the tracers. Default Tracer Config is
+ * top-level configuration option and is typically used by default tracer. Non-default tracers
+ * typically use tracer Config object.
+ */
+public record TraceOptions(
+    String tracerType, DefaultTracerConfig defaultTracerConfig, Map<String, Object> tracerConfig) {
+  private static final DefaultTracerConfig DEFAULT_TRACER_CONFIG =
+      new DefaultTracerConfig(true, false, true);
+
+  /**
+   * Default tracer configuration. Used by trace_ and debug_ methods when no tracer type is
+   * specified.
+   */
+  public static final TraceOptions DEFAULT = new TraceOptions("", DEFAULT_TRACER_CONFIG, Map.of());
+
+  /**
+   * Constructor for TraceOptions.
+   *
+   * @param tracerType the type of tracer to use
+   * @param defaultTracerConfig the default tracer's configuration
+   * @param tracerConfig the tracer configuration options for non-default tracers
+   */
+  public TraceOptions(
+      final String tracerType,
```

Comment:
an enum for the types would be nicer

Reply:
I originally had enum value for tracer type, however, it restricts the types to the one that we declared. Having a string tracer type opens it up for "plugin" based tracers in future.

Reply:
I've reintroduced TracerType as enum.